### PR TITLE
title123

### PR DESCRIPTION
**Pull request by iasemenov**
1

### CODE REVIEW

File: pushtest [hunk 1]
```diff
@@ -2,6 +2,8 @@ test
 prtestsdf
 testtesttest
 test
+abrivalg
```
Comment (iasemenov) [inline pushtest]: added

File: pushtest [hunk 2]
```diff
@@ -9,6 +9,4 @@ test
 test
 test
 12345
-12345
-12345
 12345
```
Comment (iasemenov) [inline pushtest]: delete_

File: pushtest [hunk 3]
```diff
@@ -12,5 +12,3 @@ test
...
 first
 second
-third
-forth
```
Comment (iasemenov) [inline pushtest]: removed forth line

File: pushtest [hunk 4]
```diff
@@ -15,3 +15,24 @@ second
 раз
 два
 три
+4
```
Comment (iasemenov) [inline pushtest]: первая строка последнего коммита

File: pushtest [hunk 5]
```diff
@@ -1,7 +1,38 @@
...
-test
\ No newline at end of file
+12345
+12345
+first
+second
+раз
+два
+три
+4
```
Comment (iasemenov) [inline pushtest]: первая строка последнего коммита 2

File: pushtest [hunk 6]
```diff
@@ -44,6 +44,3 @@ second
 29
 30
 31
-32
-33
-34
```
Comment (iasemenov) [inline pushtest]: remove end 3

**Comment by iasemenov** [comment]
testcomment

**Review by iasemenov — COMMENTED** [review]
123

File: pushtest [hunk 7]
```diff
@@ -1,7 +1,46 @@
 test
+фывавыап
+test
 prtestsdf
 testtesttest
 test
+abrivalg
+abrivalg
+test
 test
```
[the remark this answers is not in the document]
Comment (iasemenov) [reply pushtest:11]: test

File: pushtest [hunk 8]
```diff
@@ -1,7 +1,47 @@
...
+фывавыап
+test
 prtestsdf
 testtesttest
```
Comment (iasemenov) [inline pushtest:5]: test

File: pushtest [hunk 9]
```diff
@@ -1,7 +1,47 @@
 test
```
Comment (iasemenov) [inline pushtest:1]: review

File: pushtest [hunk 10]
```diff
@@ -1,7 +1,47 @@
...
+фывавыап
+test
 prtestsdf
```
Comment (iasemenov) [inline pushtest:4]: testcomment

File: pushtest [hunk 11]
```diff
@@ -4,8 +4,6 @@ testtesttest
 test
 abrivalg
 abrivalg
-abrivalg
-abrivalg
```
[the remark this answers is not in the document]
Comment (iasemenov) [reply pushtest]: reply no pending

[the remark this answers is not in the document]
Comment (iasemenov) [reply pushtest]: reply no pending 2

[the remark this answers is not in the document]
Comment (iasemenov) [reply pushtest]: reply no pending 3

**Review by iasemenov — COMMENTED** [review]
113

**Review by iasemenov — COMMENTED** [review]
test2aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa1233

**Comment by iasemenov** [comment]
test

**Comment by iasemenov** [comment]
test2

**Comment by iasemenov** [comment]
test delete

**Comment by iasemenov** [comment]
cmnt

**Comment by iasemenov** [comment]
testcmnt

**Comment by iasemenov** [comment]
delete

**Comment by iasemenov** [comment]
dlt

**Comment by iasemenov** [comment]
cmnt

**Comment by iasemenov** [comment]
comment

File: pushtest [hunk 12]
```diff
@@ -1,7 +1,47 @@
 test
+фывавыап
+test
 prtestsdf
 testtesttest
 test
+abrivalg
+abrivalg
```
Comment (iasemenov) [inline pushtest:8]:
```suggestion
abri
```

File: pushtest [hunk 13]
```diff
@@ -1,7 +1,47 @@
 test
+фывавыап
+test
 prtestsdf
 testtesttest
 test
+abrivalg
```
Comment (iasemenov) [inline pushtest:7]:
```
somecode
```

File: pushtest [hunk 14]
```diff
@@ -1,7 +1,47 @@
 test
+фывавыап
+test
```
Comment (iasemenov) [inline pushtest:3]: ```somecode```

Comment (iasemenov) [reply pushtest:3]: >somecode
incorrect

**Review by iasemenov — COMMENTED** [review]
123

**Comment by iasemenov** [comment]
comment

File: pushtest (hunk 8 above)
Comment (iasemenov) [inline pushtest:5]: file comment

**Review by iasemenov — COMMENTED** [review]
another review 1231234

File: pushtest (hunk 13 above)
Comment (iasemenov) [inline pushtest:7]: comment review

File: pushtest [hunk 15]
```diff
@@ -1,7 +1,47 @@
...
 prtestsdf
 testtesttest
 test
+abrivalg
+abrivalg
+asdf
```
Comment (iasemenov) [inline pushtest:9]: fiel comment from browser

Comment (iasemenov) [reply pushtest:9]: test

File: pushtest [hunk 16]
```diff
@@ -2,6 +2,10 @@ test
 prtestsdf
```
[the remark this answers is not in the document]
Comment (iasemenov) [reply pushtest:4]: test

[the remark this answers is not in the document]
Comment (iasemenov) [reply pushtest:4]: asdfasdf

File: pushtest [hunk 17]
```diff
@@ -1,7 +1,47 @@
...
+12
+13
+14
+15
+16
+17
+18
+19
+20
+21
```
Comment (iasemenov) [inline pushtest:37]: tst

Comment (iasemenov) [inline pushtest:37]: tst

File: pushtest [hunk 18]
```diff
@@ -1,7 +1,47 @@
...
+test
 test
 test
-test
\ No newline at end of file
+12345
+12345
+first
+second
+раз
```
Comment (iasemenov) [inline pushtest:17]: tst

File: pushtest [hunk 19]
```diff
@@ -1,7 +1,47 @@
...
 prtestsdf
 testtesttest
 test
+abrivalg
+abrivalg
+asdf
+test
 test
 test
-test
```
Comment (iasemenov) [inline pushtest:7]: on removed lines 2

File: pushtest [hunk 20]
```diff
@@ -1,7 +1,47 @@
...
\ No newline at end of file
+12345
+12345
+first
+second
+раз
+два
+три
```
Comment (iasemenov) [inline pushtest:19]: 123

File: pushtest [hunk 21]
```diff
@@ -1,7 +1,47 @@
...
+7
+8
+9
+10
+11
+12
+13
+14
+15
+16
```
Comment (iasemenov) [inline pushtest:32]: 123

File: pushtest [hunk 22]
```diff
@@ -1,7 +1,47 @@
...
+17
+18
+19
+20
+21
+22
```
Comment (iasemenov) [inline pushtest:38]: 123

File: pushtest [hunk 23]
```diff
@@ -1,7 +1,47 @@
...
+раз
+два
+три
+4
+5
+6
+7
+8
+9
+10
```
Comment (iasemenov) [inline pushtest:26]: 123

File: pushtest [hunk 24]
```diff
@@ -1,7 +1,47 @@
...
+12345
+12345
+first
+second
+раз
+два
+три
+4
+5
+6
```
Comment (iasemenov) [inline pushtest:22]: 123

Comment (Naid405) [reply pushtest:22]:
```java
@NotNull
  public static IntPair getGutterArea(@NotNull Editor editor, int xOffset) {
    EditorGutterComponentEx gutter = ((EditorEx)editor).getGutterComponentEx();
    int x = xOffset + 1; // leave 1px for brace highlighters
    if (ExperimentalUI.isNewUI()) {
      x += gutter.getExtraLineMarkerFreePaintersAreaOffset();
      x += 2; //IDEA-286352
      return new IntPair(x, x + (int)(JBUIScale.scale(JBUI.getInt("Gutter.VcsChanges.width", 4) * getEditorScale(editor))));
    }
    else {
      x += gutter.getLineMarkerFreePaintersAreaOffset();
    }
    int endX = gutter.getWhitespaceSeparatorOffset();
    return new IntPair(x, endX);
  }

  public static boolean isInsideMarkerArea(@NotNull MouseEvent e) {
    final EditorGutterComponentEx gutter = (EditorGutterComponentEx)e.getComponent();
    return gutter.isInsideMarkerArea(e);
  }
```

File: pushtest [hunk 25]
```diff
@@ -1,7 +1,47 @@
...
+7
+8
+9
+10
+11
```
[the remark this answers is not in the document]
Comment (Naid405) [reply pushtest:27]: 123

**Review by Naid405 — COMMENTED** [review]
???1

**Review by Naid405 — APPROVED** [review]
123

**Review by Naid405 — CHANGES_REQUESTED** [review]
123

**Comment by iasemenov** [comment]
123

**Comment by iasemenov** [comment]
123124

**Comment by iasemenov** [comment]
123

File: pushtest (hunk 8 above)
Comment (iasemenov) [reply pushtest:5]: 123

Comment (iasemenov) [reply pushtest:5]: 123123

File: pushtest (hunk 21 above)
Comment (iasemenov) [reply pushtest:32]: 123

Comment (iasemenov) [reply pushtest:32]: 12345

File: pushtest (hunk 8 above)
Comment (Naid405) [reply pushtest:5]: 123

Comment (Naid405) [reply pushtest:5]: 123

Comment (Naid405) [reply pushtest:5]: 123

File: pushtest [hunk 26]
```diff
@@ -2,6 +2,10 @@ test
 prtestsdf
 testtesttest
 test
```
[the remark this answers is not in the document]
Comment (iasemenov) [reply pushtest:6]: 123

File: pushtest (hunk 18 above)
Comment (iasemenov) [inline pushtest:17]: 123

File: pushtest [hunk 27]
```diff
@@ -1,7 +1,47 @@
...
+asdf
+test
 test
 test
-test
\ No newline at end of file
+12345
+12345
+first
+second
```
Comment (iasemenov) [inline pushtest:16]: 123

File: pushtest (hunk 8 above)
Comment (iasemenov) [reply pushtest:5]: 123

File: pushtest (hunk 14 above)
Comment (iasemenov) [inline pushtest:3]: 123

File: pushtest [hunk 28]
```diff
@@ -1,7 +1,47 @@
...
+abrivalg
+abrivalg
+asdf
+test
 test
 test
-test
\ No newline at end of file
+12345
+12345
```
Comment (iasemenov) [inline pushtest:14]: 123

File: pushtest (hunk 26 above)
[the remark this answers is not in the document]
Comment (iasemenov) [reply pushtest:6]: 123

File: pushtest (hunk 14 above)
Comment (iasemenov) [inline pushtest:3]: 123

File: pushtest (hunk 10 above)
Comment (iasemenov) [inline pushtest:4]: 123

**Pull request closed without merging**